netbox/dcim/api/views.py

```diff
@@ -396,13 +396,29 @@ def napalm(self, request, pk):
         napalm_methods = request.GET.getlist('method')
         response = OrderedDict([(m, None) for m in napalm_methods])
         ip_address = str(device.primary_ip.address.ip)
+        username = settings.NAPALM_USERNAME
+        password = settings.NAPALM_PASSWORD
         optional_args = settings.NAPALM_ARGS.copy()
         if device.platform.napalm_args is not None:
             optional_args.update(device.platform.napalm_args)
+
+        # Update NAPALM parameters according to the request headers
+        for header in request.headers:
+            if header[:7].lower() != 'napalm-':
```

Comment:
Waiting on feedback right now to see if we should opt for a `X-` prefix on the header names. (We've used custom headers elsewhere in NetBox without the prefix, but I wanted to give it some more thought.)

Reply:
Now that you mention, yea, it should be additionally prefixed with `X-`. I didn't think of that.